x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -335,41 +255,31 @@ export class SearchSessionService implements ISessionService {
   };
 
   /**
-   * Tracks the given search request/search ID in the saved session (if it exists). Otherwise, just
-   * store it in memory until a saved session exists.
+   * Tracks the given search request/search ID in the saved session.
    * @internal
    */
   public trackId = async (
     searchRequest: IKibanaSearchRequest,
     searchId: string,
-    { sessionId, isStored, strategy }: ISearchOptions,
+    { sessionId, strategy }: ISearchOptions,
```

Comment:
I don't really see any need for `isStored` any more, do you?

Reply:
😮 if so, would be great to simplify UI also

Reply:
We use this state on the client, so I didn't want to refactor it out. I don't see the harm in keeping it around atm.

Reply:
As I understand, we can leave it as the implementation detail of the client-side state session and remove a bunch of code that passes/validates this parameter
IMO, not in this pr, but would be nice to clean this up later.